app/main/views/index.py

```diff
@@ -32,6 +37,13 @@ def download_document_download(service_id, document_id):
     if not key:
         abort(404)
 
+    user_id = session.get('user_id')
+
+    # the user hasn't clicked through the landing page - redirect them back there
+    if current_user.get_id() != user_id:
+        url = url_for('main.download_document_landing', service_id=service_id, document_id=document_id, key=key)
+        return redirect(url)
```

Comment:
As far as I understand, Chris wanted to prevent users from modifying the document URL to download other files without visiting the landing page, so I think the session check should be on the actual download request, not on the page that links to it.

Reply:
Yes agree it should be there as well as here I think its good here as well so that users have to visit the landing page and get any welcome message and other information we might want to put there.